### DISCUSSION TITLE
Re-installing MOOSE Conda environment: Installation taking 1hr+ to solve environment

### DISCUSSION THREAD

**Discussion by kurtenkera** [idaholab/moose#26885] [Q&A Getting Started]
I am re-installing the MOOSE conda environment following the steps described online (link: https://mooseframework.inl.gov/getting_started/installation/conda.html). I am re-installing MOOSE to try to resolve the issue that I have brought up in this discussion post (link: https://github.com/idaholab/moose/discussions/26884). 

However, I am noticing that `conda` has taken over an hour to try to 'solve the environment'. I have pasted the terminal output below: 

```
conda create -n moose moose-dev=2024.01.23
Collecting package metadata (current_repodata.json): done
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: /
```

Can anyone help me speed up the re-installation process via `conda`?

**Comment by GiudGiud** [comment]
Hello

You need to update conda first, to get the new dependency resolver.
```
conda update conda
```
then it should run a lot faster, and you can ```conda update all``` then download moose-dev

**Reply by kurtenkera** [reply]
Thanks! Do you know how many GB of internet these updates will use?

**Reply by GiudGiud** [reply]
no i dont. Though conda will use cached (already downloaded) packages when it can iirc

**Reply by kurtenkera** [reply]
This is the output that I get when I try `conda update conda`:

```
(base) enk001@drift-ph:~$ conda update conda
Collecting package metadata (current_repodata.json): done
Solving environment: / 
The environment is inconsistent, please check the package plan carefully
The following packages are causing the inconsistency:

  - conda-forge/noarch::pyarrow-hotfix==0.6=pyhd8ed1ab_0
  - defaults/linux-64::pyarrow==14.0.2=py311hb6e97c4_0
  - defaults/linux-64::grpc-cpp==1.48.2=he1ff14a_1
  - defaults/linux-64::arrow-cpp==14.0.2=h374c478_1
  - conda-forge/linux-64::abseil-cpp==20211102.0=h93e1e8c_3
  - conda-forge/linux-64::libabseil==20211102.0=cxx17_h48a1fff_3
  - defaults/linux-64::_anaconda_depends==2024.02=py311_mkl_1
  - conda-forge/noarch::streamlit==1.31.1=pyhd8ed1ab_0
  - conda-forge/noarch::dask==2024.2.0=pyhd8ed1ab_0
failed with repodata from current_repodata.json, will retry with next repodata source.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): | WARNING conda.models.version:get_matcher(542): Using .* with relational operator is superfluous and deprecated and will be removed in a future version of conda. Your spec was 1.6.0.*, but conda is ignoring the .* and treating it as 1.6.0
WARNING conda.models.version:get_matcher(542): Using .* with relational operator is superfluous and deprecated and will be removed in a future version of conda. Your spec was 1.8.0.*, but conda is ignoring the .* and treating it as 1.8.0
WARNING conda.models.version:get_matcher(542): Using .* with relational operator is superfluous and deprecated and will be removed in a future version of conda. Your spec was 1.9.0.*, but conda is ignoring the .* and treating it as 1.9.0
WARNING conda.models.version:get_matcher(542): Using .* with relational operator is superfluous and deprecated and will be removed in a future version of conda. Your spec was 1.7.1.*, but conda is ignoring the .* and treating it as 1.7.1
done
Solving environment: / 
```

It seems there are 9 packages that are causing an 'inconsistent' base environment:  `pyarrow-hotfix`, `pyarrow`, `grpc-cpp`, `arrow-cpp`, `abseil-cpp`, `libabseil`, `_anaconda_depends`, `streamlit` and `dask`. Should I try updating these packages individually and then retry `conda update conda`? Or remove some combination of them and retry `conda update conda`?

**Reply by GiudGiud** [reply]
I would delete the conda folder and re-install it.
https://mooseframework.inl.gov/getting_started/installation/conda.html

**Reply by kurtenkera** [reply]
I should mention that prior to installing MOOSE, I had the Anaconda distribution installed. I then followed the installation steps outlined online (link: https://mooseframework.inl.gov/getting_started/installation/conda.html), but in hindsight I don't think the MiniForge installation steps below really did anything given I already had conda installed via Anaconda:

```
curl -L -O https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-Linux-x86_64.sh
bash Miniforge3-Linux-x86_64.sh -b -p ~/miniforge

export PATH=$HOME/miniforge/bin:$PATH

conda init --all
exit
```

But I made sure that I had the following channels activated nonetheless:

```
channels:
  - https://conda.software.inl.gov/public
  - conda-forge
  - defaults
```

Is this okay? I want to have Anaconda or Miniconda pre-installed as they offer a lot of packages that I can use in other environments (e.g. I have environments for training neural networks and doing other data-science things), and then I just want to create the additional 'moose' environment for MOOSE? Or is it vital to have MIniForge installed as recommended here:  https://mooseframework.inl.gov/getting_started/installation/conda.html?

**Reply by milljm** [reply]
Anaconda is fine, but you'll have to put up with Anaconda's very slow solver. This issue is why we moved away from Anaconda, to Miniforge.

If you neural network/data-science packages are available on `conda-forge`, then Miniforge can replace Anaconda. You can determine if that is the case by activating your neural/science environment, and performing a `conda list`. If you see a bunch of `conda-forge`, then you are good!
